### PR TITLE
feat: add OR join doc back

### PR DESCRIPTION
**Pull request by jithine**
## Context

With work done for https://github.com/screwdriver-cd/screwdriver/issues/2071 it's now possible to have OR join without side effects of multiple builds for same job.

## References

https://github.com/screwdriver-cd/screwdriver/issues/874

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: docs/user-guide/configuration/workflow.md [hunk 1]
```diff
@@ -13,6 +13,9 @@ toc:
     - title: Join/Fan-in
       url: "#advanced-logic-and"
       subitem: true
+    - title: Join/OR
```
Comment (tkyi) [inline docs/user-guide/configuration/workflow.md]: It's not a "join", so should think of a different term

**Review by tkyi — COMMENTED** [review]
It's kind of confusing that there are two sections for similar topics: 
- "Join/Fan-in" and "Parallel and Join"
- "Join/OR" and "Remote Triggers"
We might want to rethink organization

Also, might be good to have one example with both AND and OR logic

**Comment by adong** [comment]
Nice!

**Review by tkyi — APPROVED** [review]

**Comment by tkyi** [comment]
Not sure if you want to mention that AND and OR logic can be combined or not

**Pull request merged**